src/Java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntity_AlloyBlastSmelter.java

```diff
@@ -197,6 +198,14 @@ else if (isUsingControllerCircuit  && tInputList.size() >= 1) {
 					}
 					this.mMaxProgresstime = Math.max(1, this.mMaxProgresstime);
 					this.mOutputFluids = new FluidStack[]{tRecipe.getFluidOutput(0)};
+					List<ItemStack> tOutPutItems = new ArrayList<ItemStack>();
+					for (ItemStack tOut : tRecipe.mOutputs) {
+						if (tOut != null) {
```

Comment:
Can you use ItemUtils.isItemValid()?
Better validity check vs a flat null check.